src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -33,49 +43,59 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
 
             _dbgStdInName = UnixUtilities.MakeFifo(Logger);
             _dbgStdOutName = UnixUtilities.MakeFifo(Logger);
-
-            string watcherDirectory = _dbgStdInName.Substring(0, _dbgStdInName.IndexOf(UnixUtilities.FifoPrefix, StringComparison.OrdinalIgnoreCase));
-            _fifoWatcher = new FileSystemWatcher(watcherDirectory, UnixUtilities.FifoPrefix + "*");
-            _fifoWatcher.Deleted += FifoWatcher_Deleted;
-            _fifoWatcher.EnableRaisingEvents = true;
+            string pidFifo = UnixUtilities.MakeFifo(Logger);
+            string exitFifo = UnixUtilities.MakeFifo(Logger);
 
             // Setup the streams on the fifos as soon as possible.
             FileStream dbgStdInStream = new FileStream(_dbgStdInName, FileMode.Open);
             FileStream dbgStdOutStream = new FileStream(_dbgStdOutName, FileMode.Open);
+            FileStream pidStream = new FileStream(pidFifo, FileMode.Open);
+            FileStream exitStream = new FileStream(exitFifo, FileMode.Open);
 
             string debuggerCmd = UnixUtilities.GetDebuggerCommand(localOptions);
             string launchDebuggerCommand = UnixUtilities.LaunchLocalDebuggerCommand(
                 debuggeeDir,
-                debuggerCmd,
+                exitFifo,
                 _dbgStdInName,
-                _dbgStdOutName
-                );
+                _dbgStdOutName,
+                pidFifo,
+                debuggerCmd);
 
             TerminalLauncher terminal = TerminalLauncher.MakeTerminal("DebuggerTerminal", launchDebuggerCommand, localOptions.Environment);
             terminal.Launch(debuggeeDir);
 
+            using (StreamReader pidReader = new StreamReader(pidStream, Encoding.UTF8, true, UnixUtilities.StreamBufferSize))
+            {
+                _debuggerPid = int.Parse(pidReader.ReadLine(), CultureInfo.InvariantCulture);
```

Comment:
Do we need to do something to handle the terminal process failing and therefore never running our bash script? (ex: wait on the terminal process to exit and if so cancel this read?)